### PR TITLE
Add Sonarcloud Scan ci

### PR DESCRIPTION
**Pull request by fengshunli**
### Ⅱ. Does this pull request fix one issue?

fixes #XXXX

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: .github/workflows/sonarcloud.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+name: Sonarcloud Scan
+
+on:
+  push:
+    branches:
+      - 'main'
```
Comment (cheyang) [inline .github/workflows/sonarcloud.yaml:6]: I think the branch should be master instead of main.

Comment (fengshunli) [reply .github/workflows/sonarcloud.yaml]: done

**Review by TrafalgarZZZ — APPROVED** [review]
/lgtm

**Pull request merged**